dataland-frontend/src/components/forms/parts/elements/basic/UploadDocumentsFormWithComment.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <div class="form-field" :data-test="name">
+    <FormKit v-model="baseDataPoint" type="group" name="dataSource">
+      <UploadDocumentsForm
+        @updatedDocumentsSelectedForUpload="handleDocumentUpdatedEvent"
+        ref="uploadDocumentsForm"
+        :name="name"
+        :more-than-one-document-allowed="false"
+        :file-names-for-prefill="fileNamesForPrefill"
+      />
+      <FormKit type="hidden" name="fileName" v-model="documentName" />
+      <FormKit type="hidden" name="fileReference" v-model="documentReference" />
+      <FormKit
+        type="textarea"
+        name="comment"
+        placeholder="(Optional) Add comment that might help Quality Assurance to approve the datapoint. "
+      />
+    </FormKit>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import { BaseFormFieldProps } from "@/components/forms/parts/fields/FormFieldProps";
+import UploadDocumentsForm from "@/components/forms/parts/elements/basic/UploadDocumentsForm.vue";
+import { type DocumentToUpload } from "@/utils/FileUploadUtils";
+import { type BaseDataPoint } from "@/utils/DataPoint";
+
+export default defineComponent({
+  name: "UploadDocumentsFormWithComment",
+  components: { UploadDocumentsForm },
+  inheritAttrs: false,
+  props: { ...BaseFormFieldProps },
+  data() {
+    return {
+      baseDataPoint: {} as BaseDataPoint<unknown>,
+      referencedDocument: undefined as DocumentToUpload | undefined,
+      documentName: "",
+      documentReference: "",
```

Comment:
It should be avoided to introduce empty strings for fileReferences and fileNames

Reply:
Totally agree.   @MarcinStanczyk7  you should make sure that the fileReference is never empty!
Furthermore, the name of this file ("UploadDocumentsFormWithComment") is misleading.
This is a list of BaseDataPointStrings. There are no comments here!

Reply:
@MarcinStanczyk7  can you have a look at this comment here?

Reply:
@emanueldfine He certainly won't in time.

Reply:
My commit (commitId = 7f71ca8b... `avoid possibility of empty-fileReference-uploads by throwing Error + some refactoring`) takes care of these issues:

1.) I renamed this vue-component to make it clear what it is being used for.
2.) Before the event-emit, I am assuring that fileReference is defined. If it is undefined, an Error is thrown. Therefore it is absolutely impossible now, that a Gdv dataset is being uploaded with an empty fileReference anywhere.